src/lib/tabs/tab-body.ts

```diff
@@ -132,46 +143,43 @@ export class MatTabBody implements OnInit {
   /** The tab body content to display. */
   @Input('content') _content: TemplatePortal;
 
+  /** Position that will be used when the tab is immediately becoming visible after creation. */
+  @Input() origin: number;
+
   /** The shifted index position of the tab body, where zero represents the active center tab. */
   @Input()
   set position(position: number) {
-    if (position < 0) {
-      this._position = this._getLayoutDirection() == 'ltr' ? 'left' : 'right';
-    } else if (position > 0) {
-      this._position = this._getLayoutDirection() == 'ltr' ? 'right' : 'left';
-    } else {
-      this._position = 'center';
-    }
+    this._positionIndex = position;
+    this._computePositionAnimationState();
   }
-  _position: MatTabBodyPositionState;
-
-  /** The origin position from which this tab should appear when it is centered into view. */
-  @Input()
-  set origin(origin: number) {
-    if (origin == null) { return; }
 
-    const dir = this._getLayoutDirection();
-    if ((dir == 'ltr' && origin <= 0) || (dir == 'rtl' && origin > 0)) {
-      this._origin = 'left';
-    } else {
-      this._origin = 'right';
+  constructor(private _elementRef: ElementRef,
+              @Optional() private _dir: Directionality,
+              // TODO(paul): make the changeDetectorRef required when doing breaking changes.
```

Comment:
Mark this with a `@deletion-target`, otherwise we'll miss it when doing breaking changes.

Reply:
I wasn't sure about that because `deletion-target` kind of implies that we are going to delete that parameter.